src/Network/Broadcast/OutboundQueue.hs

```diff
@@ -440,8 +443,19 @@ data OutboundQ msg nid = forall self .
 
       -- | Signal we use to wake up blocked threads
     , qSignal :: Signal CtrlMsg
+
+      -- | Mutable classification of peers.
+    , qClassifyPeers :: MVar (Classification nid)
```

Comment:
Oh, it's a new MVar. Hmm, somehow I thought this would be integrated with the subscriptions. I need to think about this patch more carefully.